enhancements/image-references.md

```diff
@@ -0,0 +1,241 @@
+
+The only acceptable process for creating an operator bundle will be to run the
+`operator-sdk generate bundle` CLI, providing the manifests, metadata and image
+sources as input to the creation process.
+
+### Proposed UX
+
+A set of new files in the `metadata/` directory of a bundle will be authored by
+the manifest authors and build systems to record image replacement intent and
+execution.
+
+Operator manifest authors write a manifest that refers to images using some
+opaque string, and provide an `image-references.yaml` file alongside their
+manifests that declares which strings inside of their manifest are referring
+to pull specifications of images and names each occurence.
+
+The `metadata/image-references.yaml` file holds data in the following format, mapping
+common names of container images to their string placeholders in the manifests:
+
+```yaml
+imageReferences:
+- name: common-name
+  substitute: registry.svc.ci.openshift.org/openshift:image
+```
+
+This mapping, therefore, defines what needs to be replaced in manifests when
+they are bundled and identifies each replacement with a name. When
+`operator-sdk generate bundle` runs, it will require as input a second mapping
+from those names to literal image pull specifications and will run the replacement.
+In this manner, the configuration provided by the manifest author remains static
+regardless of the eventual replacement that a build system will execute.
+
+The second mapping required at build-time, known as `image-replacements.yaml`, will take
+a similar form, mapping the common names of images to their explicitly resolved pull
+specifications. This file will have the following format:
+
+```yaml
+imagePullSpecs:
+- name: common-name
+  pullSpec: quay.io/openshift-release-dev/ocp-v4.0-art-dev@sha256:298d0b496f67a0ec97a37c6e930d76db9ae69ee6838570b0cd0c688f07f53780
+```
+
+`operator-sdk generate bundle` will use this chain of mapping to perform replacements
+in the manifests before creating a bundle image layer. The layer creation will also
+be shared logic with `oc adm release new` in order to allow both processes to build
+image layers without requiring the use of a container runtime, other build system, any
+elevated permissions, privileges, capacities or SELinux roles. As the output image
+layer in both cases is `FROM scratch` and simply contains manifest data, this build
+process is simple and producing the layer by creating the underlying tar bundle does
+not come with risks.
+
+The bundle creation process will commit the `image-replacements.yaml` file into the output
+bundle, and will furthermore create as output a `release-metadata.yaml` metadata file
+that will expose any further build-time inputs used to create the bundle, so that downstream
+consumers can access this data as necessary. The format for the `release-metadata.yaml` file
+is a loose key-value store:
+
+```yaml
+metadata:
+  some-key: some-value
+```
```

Comment:
Understand this is intended as open-ended metadata, but providing examples of what a build system might include here would provide more context.